cpp/src/io/csv/csv_writer.cu

```diff
@@ -50,23 +51,51 @@ NVStrings* column_to_strings_csv(const gdf_column* column, gdf_size_type row_off
     auto valid = column->valid;
     if( valid )                                    // normalize row_offset (number of bits here)
         valid += (row_offset / GDF_VALID_BITSIZE); // to appropriate pointer for the bitmask
+
+    auto gdf2nvs_timeunit = [] (gdf_time_unit time_unit) -> NVStrings::timestamp_units {
+        if( time_unit==TIME_UNIT_s )
+            return NVStrings::seconds;
+        if( time_unit==TIME_UNIT_us )
+            return NVStrings::us;
+        if( time_unit==TIME_UNIT_ns )
+            return NVStrings::ns;
+        return NVStrings::ms;
+    };
+
     switch( column->dtype )
     {
         case GDF_STRING:
             rtn = (static_cast<NVStrings*>(column->data))->sublist(row_offset,row_offset+rows);
             break;
+        case GDF_STRING_CATEGORY:
```

Comment:
This should use the `type_dispatcher` https://github.com/rapidsai/cudf/blob/branch-0.9/cpp/src/utilities/type_dispatcher.hpp#L118

Reply:
The switch statement?

Reply:
The type-dispatcher does not handle `GDF_STRING` type.
I could if around that type.
Is there a tutorial on how to use the dispatcher for this? Seems there is not much common code between the types. I would be just moving the code into separate functors, right?

Reply:
Correct, it does not yet support `GDF_STRING`. In that case, I would do:
```
 if(dtype == GDF_STRING){ // do stuff}
else{
   cudf::type_dispatcher(dtype, functor{}, ...);
}
``` 

I opened a PR with extra documentation in the `type_dispatcher` for specializing dispatched functors: https://github.com/rapidsai/cudf/pull/2193

Reply:
Still seems like I would just be writing unique individual functors for each of the types since the calls to NVStrings are all unique. Does that sound right?

Reply:
That depends. 

Can the NVStrings function be made into a template such that you can just invoke the same function for all the different types? 

Even if you end up with separate specializations for each type, this should still use the type_dispatcher for consistency. It ensures that we fail in a consistent way for an unsupported type, and also helps to future proof against adding new types in the future.

Reply:
Changed switch statement to type-dispatcher call.

Reply:
Beautiful.